File: source/includes/read/distinct.php
<?php
require 'vendor/autoload.php';

use MongoDB\BSON\Document;

$uri = getenv('MONGODB_URI') ?: throw new RuntimeException('Set the MONGODB_URI variable to your Atlas URI that connects to the sample dataset');
$client = new MongoDB\Client($uri);

// start-db-coll
$collection = $client->sample_restaurants->restaurants;
// end-db-coll

// Retrieves distinct values of the "borough" field
// start-distinct
$results = $collection->distinct('borough', []);
foreach ($results as $value) {
    echo json_encode($value), PHP_EOL;
}
// end-distinct

// Retrieves distinct "borough" field values for documents with a "cuisine" value of "Italian"
// start-distinct-with-query
$results = $collection->distinct('borough', ['cuisine' => 'Italian']);
foreach ($results as $value) {
    echo json_encode($value), PHP_EOL;
}
// end-distinct-with-query

// Retrieves distinct "name" field values for documents matching the "borough" and "cuisine" fields query
// and attaches a comment to the operation
// start-distinct-with-comment
$query = ['borough' => 'Bronx', 'cuisine' => 'Pizza'];
$options = ['comment' => 'Bronx pizza restaurants'];
$results = $collection->distinct('name', $query, $options);

foreach ($results as $value) {
    echo json_encode($value), PHP_EOL;
}
// end-distinct-with-comment

File: content/golang/v2.1/source/includes/usage-examples/code-snippets/distinct.go
// Retrieves distinct values of a field by using the Go driver
package main

import (
	"context"
	"fmt"
	"log"
	"os"

	"github.com/joho/godotenv"
	"go.mongodb.org/mongo-driver/v2/bson"
	"go.mongodb.org/mongo-driver/v2/mongo"
	"go.mongodb.org/mongo-driver/v2/mongo/options"
)

type Restaurant struct {
	ID           bson.ObjectID `bson:"_id"`
	Name         string
	RestaurantId string `bson:"restaurant_id"`
	Cuisine      string
	Address      interface{}
	Borough      string
	Grades       interface{}
}

func main() {
	if err := godotenv.Load(); err != nil {
		log.Println("No .env file found")
	}

	var uri string
	if uri = os.Getenv("MONGODB_URI"); uri == "" {
		log.Fatal("You must set your 'MONGODB_URI' environment variable. See\n\t https://www.mongodb.com/docs/drivers/go/current/usage-examples/#environment-variable")
	}

	client, err := mongo.Connect(options.Client().ApplyURI(uri))
	if err != nil {
		panic(err)
	}
	defer func() {
		if err = client.Disconnect(context.TODO()); err != nil {
			panic(err)
		}
	}()

	// Filters the collection for documents where the value of cuisine is "Tapas"
	coll := client.Database("sample_restaurants").Collection("restaurants")
	filter := bson.D{{"cuisine", "Tapas"}}

	// Retrieves the distinct values of the "borough" field in documents
	// that match the filter
	var arr []string
	err = coll.Distinct(context.TODO(), "borough", filter).Decode(&arr)
	if err != nil {
		panic(err)
	}

	// Prints the distinct "borough" values
	for _, result := range arr {
		fmt.Println(result)
	}

	// When you run this file, it should print:
	// Brooklyn
	// Manhattan
	// Queens
}
